FirebaseAuth/Sources/AuthProvider/Phone/FIRPhoneAuthProvider.m

```diff
@@ -739,21 +743,43 @@ - (void)reCAPTCHAURLWithEventID:(NSString *)eventID completion:(FIRReCAPTCHAURLC
-                                                                     value:self->_auth
-                                                                               .requestConfiguration
-                                                                               .languageCode]];
-                                     }
                                      NSURLComponents *components = [[NSURLComponents alloc]
                                          initWithString:
                                              [NSString stringWithFormat:kReCAPTCHAURLStringFormat,
                                                                         authDomain]];
                                      [components setQueryItems:queryItems];
-                                     if (completion) {
-                                       completion([components URL], nil);
+                                     NSURL *url = [components URL];
+                                     if (appCheck) {
+                                       [appCheck
+                                           getTokenForcingRefresh:false
+                                                       completion:^(
+                                                           id<FIRAppCheckTokenResultInterop> _Nonnull tokenResult) {
+                                                         if (tokenResult.error) {
+                                                           FIRLogWarning(
+                                                               kFIRLoggerAuth, @"I-AUT000018",
+                                                               @"Error getting App Check token; "
+                                                               @"using placeholder token "
+                                                               @"instead. Error: %@",
+                                                               tokenResult.error);
+                                                         }
+                                                         NSString *appCheckTokenFragments = [@"fac="
+                                                             stringByAppendingString:tokenResult
+                                                                                         .token];
+                                                         NSString *URLString = [url absoluteString];
+                                                         NSString *URLStringWithAppCheckToken =
+                                                             [URLString stringByAppendingString:
+                                                                            appCheckTokenFragments];
+                                                         NSURL *URLWithAppCheckToken =
+                                                             [NSURL URLWithString:
+                                                                        URLStringWithAppCheckToken];
```

Comment:
I took a look at the `NSURLComponents` [docs](https://developer.apple.com/documentation/foundation/nsurlcomponents?language=objc) -- I think you should be able to call `setFragment` directly on the components object, rather than having to transform it into an `NSString` and then back into an `NSURL`.

(FIROAuthProvider.m should also probably do this as well, but since that's existing behavior and would need a larger refactor, you can just make the change in this file for now)

Reply:
yes, thx for checking that! Have corrected.